backend/msgpack/queue_test.go

```diff
@@ -51,16 +51,19 @@ func TestInstanceQueueEnqueueQueueFull(t *testing.T) {
 	opts := testServerOptions().SetInstanceQueueSize(1)
 	queue := newInstanceQueue(testPlacementInstance, opts).(*queue)
 
-	// Fill up the queue and park the draining goroutine for 5 seconds
-	// so the queue remains full.
-	queue.writeFn = func([]byte) error { time.Sleep(5 * time.Second); return nil }
-	for i := 0; i < 10; i++ {
-		select {
-		case queue.bufCh <- msgpack.NewBufferedEncoder():
-		default:
-		}
+	// Fill up the queue and park the draining goroutine so the queue remains full.
+	parkCh := make(chan struct{})
+	drainCh := make(chan struct{})
+	queue.writeFn = func([]byte) error {
```

Comment:
Hmm, I think it should be alright actually since `drain` will pull the buffer off the channel before calling `writeFn`. So the second call should be able to proceed after `drain` pulls the first buffer off the channel, but then no more buffers will be pulled off the channel since `writeFn` and therefore `drain` is blocked. Just tested it and it seemed to run fine, though the linter did complain about `return nil` being unreachable.